tests/test_cerberus_translator.py

```diff
@@ -0,0 +1,102 @@
+
+
+@pytest.mark.parametrize(
+    "error,message",
+    [
+        pytest.param(
+            "required field",
+            messages.MISSING_REQUIRED_FIELD,
+            id="Message of REQUIRED Field",
+        ),
+        pytest.param(
+            "empty values not allowed",
+            messages.EMPTY_NOT_ALLOWED,
+            id="Message of Empty Value",
+        ),
+        pytest.param(
+            "Unknown field",
+            messages.UNKNOWN_FIELD,
+            id="Message of Unknown field is present",
+        ),
+        pytest.param(
+            "null value not allowed",
+            messages.NULL_NOT_ALLOWED,
+            id="Message when NULL value is present",
+        ),
+        pytest.param(
+            "value does not match regex '^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$'",
+            messages.REGEX_NOT_MATCHED,
+            id="Message of when REGEX NOT MATCHED",
+        ),
+    ],
+)
+def test_cerberus_translator_valid_messages(error, message):
+    translator = CerberusMessageTranslator()
+    assert translator.translate_message(error) == message
+
+
+@pytest.mark.parametrize(
+    "error,message",
+    [
+        pytest.param(
+            "value does not match regex '^[a-z]*$'",
+            messages.MISSING_DEPENDENT_FIELD,
+            id="Invalid message for REGEX NOT MATCHED error",
+        ),
+        pytest.param(
+            "Unknown field",
+            messages.UNEXPECTED_FIELD,
+            id="Invalid error of UNEXPECTED FIELD",
+        ),
+    ],
+)
+def test_cerberus_translator_invalid_messages(error, message):
+    translator = CerberusMessageTranslator()
+    with pytest.raises(AssertionError):
+        assert translator.translate_message(error) == (message)
+
+
+@pytest.mark.parametrize(
+    "schema,data,valid,expected_errors",
+    [
+        pytest.param(
+            {"foo": {"nullable": False, "type": "string"}, "bar": {"type": "integer"}},
+            {"foo": None},
+            False,
+            {"foo": [messages.NULL_NOT_ALLOWED]},
+            id="Nullable - Invalid case",
+        ),
+        pytest.param(
+            {"foo": {"required": True, "type": ["binary", "list"]}},
+            {"foo": [101, 1111111]},
+            True,
+            {},
+            id="Binary, multiple types - Valid case",
+        ),
+        pytest.param(
+            {
+                "foo": {
+                    "type": "dict",
+                    "schema": {
+                        "address": {"type": "string"},
+                        "city": {"type": "string", "required": True},
+                    },
+                },
+                "1": {"empty": False, "type": "string"},
+            },
+            {"foo": {"address": "my address", "city": "my town"}, "1": ""},
+            False,
+            {"1": [messages.EMPTY_NOT_ALLOWED]},
+            id="Empty values - Invalid Case",
+        ),
+    ],
+)
+def test_required_rule(schema, data, valid, expected_errors):
```

Comment:
This test shouldn't be here, right?

Reply:
Yes @ejulio as per our previous discussion. I took some time to learn about testing concepts, tools, and best practices. I feel here in this instance these tests (Integration tests) can go well with our unit tests just because of the fact that I am testing only those messages which are added new to the `messages.py`, specific to Spidermon, and helps us debug faster when these edge-cases so as to say breaks somewhere down the line.  

These new tests actually helped me out, to debug and write better messages for the translator as well as find that `feature we have missed` that was throwing a `TypeError` (Also talked about this)

For the location, I know well now that this isn't the right spot for these tests considering their nature. But, I think they will be good here as they cover new messages, also everything related to the translator that might potentially trouble us is in just one spot and I think we can grow quite better from here on out. 

**Here's my flow**
If we add a new message to the translator, we can also add a test to test it out below. Leading to Unit test and integration going hand in hand rather than us switching files to find 2 different tests that more or less are made for the same component. What say?